ex04.ex

```diff
@@ -75,7 +87,23 @@ defmodule Ex04 do
   return value will be the thing you have to manipulate.
   """
 
-  def even_odd . . . "your code"
+  def filter(true, [ h | t ], func) do
+    [ h | filter(t, func) ]
+  end
+  def filter(false, [ _h | t ], func) do
+    filter(t, func)
+  end
+  def filter([ h | t ], func) do
+    filter(func.(h), [ h | t ], func)
+  end
+  def filter([], _func), do: []
+
```

Comment:
I love the fact that you're passing a predicate function in to your filter function. Nicely done.

Countering that is the fact that you process the list twice, once to get the evens and once for the odds.

No deduction, but can you think of a way to do it in one pass?